src/entry.rs

```diff
@@ -622,18 +636,23 @@ impl<'a> EntryFields<'a> {
             _preserve: bool,
         ) -> io::Result<()> {
             match f {
-                Some(f) => {
-                    let mut perm = f.metadata()?.permissions();
-                    perm.set_readonly(mode & 0o200 != 0o200);
-                    f.set_permissions(perm)
+                Some(_f) => {
+                    // Set at file creation
+                    Ok(())
                 }
                 None => {
                     let mut perm = r#try!(fs::metadata(dst)).permissions();
-                    perm.set_readonly(mode & 0o200 != 0o200);
+                    perm.set_readonly(readonly(&Ok(mode)));
                     fs::set_permissions(dst, perm)
                 }
             }
         }
+        #[cfg(windows)]
+        fn readonly(mode: &io::Result<u32>) -> bool {
```

Comment:
Could this change to `fn(u32) -> bool` since that's basically what's being deduplicated here?

Reply:
also could the name perhaps be something like `is_readonly_for_win32` or something like that?

Reply:
I have renamed it to user_can_write - I was staring at it and this jumped out at me; let me know if you would still prefer is_readonly_for_win32 : the function isn't actually win32 specific, its just only used on win32 today.

I don't want to change the signature though, because this cascades into the secure open function and it gets messy: that can't be a closure because there is already a closure borrowing self; as a function passing in windows specific parameters like 'readonly' will make the code hard to maintain as more os-specific cases occur, so I'd rather pass in only the actual data that is being operated on - e.g. the mode, and as mode is a monad, not a value, keeping the monad makes a lot of sense to me.

Reply:
Let's please change this to `u32 -> bool`, unnecessarily having to wrap this in `&Ok(...)` in one callsite is pretty wonky.

Reply:
I'll have a go at this, I started there and worked in to end up with the shape I got, but I'll see if I can make it not too ugly.